### DISCUSSION TITLE
How to show Activity Diagram, using Python?

### DISCUSSION THREAD

**Discussion by AlexanderScheqolev** [gaphor/gaphor#3017] [Q&A] [answered]
Hi.
I have met some questions, working with Python library.
I know, how to show and get image of a usual diagram, but how can we get an Activity Diagram?

**Comment by danyeaw** [comment, marked as answer]
Hi @AlexanderScheqolev, thanks for the question!

You can get a certain type of diagram by setting the diagram's diagramType attribute. For example, to create a UML Activity Diagram with an actvity drawn in it:

```
from gaphor import UML
from gaphor.core.modeling import Diagram, ElementFactory
from gaphor.diagram.drop import drop
from gaphor.extensions.ipython import auto_layout, draw

element_factory = ElementFactory()

act_diagram = element_factory.create(Diagram)
act_diagram.name = "Process Controller"
act_diagram.diagramType = "act"

activity = element_factory.create(UML.Activity)
activity.name = "Control Process"
drop(activity, act_diagram, 0, 0)

auto_layout(act_diagram)
draw(act_diagram, format="svg")
```

![image](https://github.com/gaphor/gaphor/assets/10014976/095c6c39-563f-4190-8bfa-e06903555764)

**Comment by AlexanderScheqolev** [comment]
Thank you for your answer @danyeaw !

However, I also want to know one thing. Can I get elements from diagram and what module i should use?

**Reply by alvaro562003** [reply]
It's so simple that it's beautiful.
